crates/wysiwyg/src/composer_model/code_block.rs

```diff
@@ -435,58 +247,97 @@ where
         self.create_update_replace_all()
     }
 
-    fn find_ancestor_to_split(&self, handle: &DomHandle) -> DomHandle {
-        let path_len = handle.depth();
-        if path_len <= 1 {
-            DomHandle::root()
-        } else {
-            for i in (0..handle.depth()).rev() {
-                let ancestor_handle = handle.sub_handle_up_to(i);
-                let ancestor = self.state.dom.lookup_node(&ancestor_handle);
-                match ancestor.kind() {
-                    Generic | List | ListItem | CodeBlock => {
-                        return ancestor_handle
-                    }
-                    _ => continue,
-                }
-            }
-            panic!("Should never reach this point, one of the parents surely can be split.");
-        }
-    }
-
-    pub(crate) fn format_node_for_code_block(node: &DomNode<S>) -> DomNode<S> {
+    pub(crate) fn format_node_for_code_block(
```

Comment:
It's a bit hard to follow what's happening in this method because of the deeply nested branches. Perhaps we could refactor a bit to flatten it and/or add some comments to explain how it works?